src/mlpack/methods/amf/init_rules/average_init.hpp

```diff
@@ -74,6 +74,77 @@ class AverageInitialization
     H = H + avgV;
   }
 
+  /**
+   * Initialize the matrix W or H to the average value of V with uniform
+   * random noise added.
+   *
+   * @param V Input matrix.
+   * @param r Rank of matrix.
+   * @param M W or H matrix, to be initialized to the average value of V
+   * with uniform random noise added.
+   * @param whichMatrix If true, initialize W. Otherwise, initialize H.
+   */
+  template<typename MatType>
+  inline static void InitializeOne(const MatType& V,
+                                   const size_t r,
+                                   arma::mat& M,
+                                   const bool whichMatrix = true)
+  {
+    const size_t n = V.n_rows;
+    const size_t m = V.n_cols;
+
+    if (whichMatrix)
+    {
+      double avgV = 0;
+      size_t count = 0;
+      double min = DBL_MAX;
+
+      // Iterate over all elements in the matrix (for sparse matrices, this only
+      // iterates over nonzeros).
+      for (typename MatType::const_row_col_iterator it = V.begin();
+          it != V.end(); ++it)
+      {
+        ++count;
+        avgV += *it;
+        // Track the minimum value.
+        if (*it < min)
+          min = *it;
```

Comment:
Can't we use `arma::min` and `arma::mean` to do the same?

Reply:
I just take a brief look at the Armadillo documentation. I think it should be ok to find the minimum by using arma::min. To find the mean value of V, I think there might be some problems because we do not know whether V is a vector or a matrix. For a vector, it's just mean(V). For a matrix, it's mean(mean(M)).

Reply:
I see, no need to do it differently.

Reply:
I suppose `mean(vectorise(V))` could work too---up to you what you want to do. :+1: